pycbc/cosmology.py

```diff
@@ -114,78 +254,145 @@ def get_redshift(self, dist):
+# set up D(z) interpolating classes for the standard cosmologies
+_d2zs = {_c: DistToZ(cosmology=_c)
+         for _c in astropy.cosmology.parameters.available}
+
 
-    Cosmological constants `h, om, ol, w0, w1, w2` may optionally be provided
-    (see below). Otherwise, standard LambdaCDM will be used by default; see
-    `LALCosmologyCalculator` for details.
+def redshift(distance, **kwargs):
+    r"""Returns the redshift associated with the given luminosity distance.
+
+    If the requested cosmology is one of the pre-defined ones in
+    :py:attr:`astropy.cosmology.parameters.available`, :py:class:`DistToZ` is
+    used to provide a fast interpolation. This takes a few seconds to setup
+    on the first call.
 
     Parameters
     ----------
     distance : float
         The luminosity distance, in Mpc.
-    h : {None, float}
-        The normalized Hubble constant to use. If None, will default to
-        standard cosmology.
-    om : {None, float}
-        The matter energy density. If None, will default to standard cosmology.
-    ol : {None, float}
-        The cosmological-constant density. If None, will default to standard
-        cosmology.
-    w0 : {None, float}
-        The 0th-order dark energy equation of state parameter. If None, will
-        default to standard cosmology.
-    w1 : {None, float}
-        The first-order dark energy equation of state parameter. If None, will
-        default to standard cosmology.
-    w2 : {None, float}
-        The second-order dark energy equation of state parameter. If None, will
-        default to standard cosmology.
+    \**kwargs :
+        All other keyword args are passed to :py:func:`get_cosmology` to
+        select a cosmology. If none provided, will use
+        :py:attr:`DEFAULT_COSMOLOGY`.
 
     Returns
     -------
     float :
         The redshift corresponding to the given distance.
     """
-    if all([p is None for p in [h, om, ol, w0, w1, w2]]):
-        # just use the default converter
-        d2z = _d2z
-    else:
-        d2z = _DistToZ(h=h, om=om, ol=ol, w0=w0, w1=w1, w2=w2)
-    return d2z(distance)
+    cosmology = get_cosmology(**kwargs)
+    try:
+        z = _d2zs[cosmology.name](distance)
+    except KeyError:
+        # not a standard cosmology, call the redshift function
+        z = _redshift(distance, cosmology=cosmology)
+    return z
+
+
+def redshift_from_comoving_volume(vc, **kwargs):
+    r"""Returns the redshift from the given comoving volume.
+
+    Parameters
+    ----------
+    vc : float
+        The comoving volume, in units of cubed Mpc.
+    \**kwargs :
+        All other keyword args are passed to :py:func:`get_cosmology` to
+        select a cosmology. If none provided, will use
+        :py:attr:`DEFAULT_COSMOLOGY`.
+
+    Returns
+    -------
+    float :
+        The redshift at the given comoving volume.
+    """
+    cosmology = get_cosmology(**kwargs)
+    # first get the redshift associated with the given comoving volume
```

Comment:
doesn't this comment belong in the next function instead?

Reply:
Thanks catching that, removed.